### DISCUSSION TITLE
Little inconvenience in token generation results in login trouble

### DISCUSSION THREAD

**Discussion by Metanormal** [community/community#14800] [Other Feature Feedback, Questions, & Ideas]
When generating “Personal access tokens” via the developer settings the resulting token seems to be not enclosed in any limitting html-tags. When i double-click on it to copy it to the clipboard it will also copy an additional space at the end of the token.

When using this token in (for me it was) Android Studio the resulting error is very misleading:

`Support for password authentication was removed on August 13, 2021.
Please use a personal access token instead.
remote: Please see [Token authentication requirements for Git operations | The GitHub Blog 1](https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/) for more information.
Authentication failed for 'https://github.com/ … ’`

I lost 4 hours until i realized the tailing space.

Of course i could have looked at the token i am copying to the clipboard
Of course i could have trimmed my token from spaces
Of course i could have used the copy-button

But also i could have gotten a simpler answer like “wrong token”
But also the token could have been trimmed before checked.

My suggestion to the github team would be to automatically trim the given token from spaces, unless leading and tailing spaces are possible or wanted. Or prevent that a double-click results in marking the token plus one space.

Also the resulting error could differ from above statement.

Just to make on thing clear: I understand, that the trouble i went thru were my own fault. :slight_smile:

I say 42, Martin Rieth
---
First mentioned here: https://github.community/t/little-inconvenience-in-token-generation-results-in-login-trouble/244297

**Comment by queenofcorgis** [comment]
Thanks for posting, @Metanormal 😄
